src/components/Table/components/RenderRowOrNestedRow/components/ContentCell/ContentCell.tsx

```diff
@@ -64,7 +66,7 @@ const ContentCell: React.FC<Props> = ({
         {isComponentFunctionType(content) ? (
           content({ content, colSpan })
         ) : (
-          <span data-column={columns[cellCounter]}>{content}</span>
+          <span data-column={col}>{content}</span>
```

Comment:
`if typeof col === ExtendedColumn` then this would take the following form:
`data-column="[object Object]"`. 
we need some type checking here as in line 58